internal/workflow.go

```diff
@@ -66,8 +66,20 @@ var (
 )
 
 type (
+	// commonChannel contains methods common to both SendChannel and ReceiveChannel
+	commonChannel interface {
+		// Name returns the name of the Channel.
+		// If the Channel was retrieved from a GetSignalChannel call, Name returns the signal name.
+		//
+		// A Channel created without an explicit name will use a generated name by the SDK and
+		// is not deterministic.
+		Name() string
+	}
+
 	// SendChannel is a write only view of the Channel
 	SendChannel interface {
+		commonChannel
```

Comment:
I wonder if this should be on the receive side only (send channel only used if user explicitly chooses to break it up, and they can't get things like `Len()` either even though _technically_ we could provide it).

Reply:
Just wondering aloud. Can ask the same of `Len`. Go allows you to get `len` and `cap` on both recv-only and send-only channels but we decided `Len()` should only be on receive. I just wonder if we need to apply whatever that thought process is here. I don't mind if not (though I think it'd be worth copying the definition twice vs making a new interface).

Reply:
For our use-case we only use `Name` for the `SendChannel` case, so if anything I'd only want it on the `SendChannel`.